[DominicFuMJ] iP

DominicFuMJ opened the pull request:
### Victor

> "Plans are worthless. Planning is essential." ~ Dwight D. Eisenhower ([source](https://www.inloox.com/company/blog/articles/the-62-most-inspiring-project-management-quotes/))

Victor helps to organize all of your upcoming events. Never again would you ever forget an important task, deadline or event. It is 

- text-based
- easy to read for ~blind~ people with impaired sight
- simple to use, even for less tech savvy people 

To use Victor, you just need to:
1. Download the program from [here](https://github.com/nus-cs2103-AY2324S2/ip/pull/324)
2. Run the program
3. Add your tasks
4. Check periodically so that this program actually has any use. 😉

And best of all, it's **FREE**!

Victor features include:

- [x] Managing Tasks
- [x] Searching for Tasks
- [ ] Reminders (coming soon)

If you are proficient in writing Java code, you may help to improve anything that you think could use improvement. The
main method is included below:
```
public static void main(String[] args) throws IOException {
        new Victor("data/victor.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Level 0. Rename, Greet, Exit

Commit:
Level 1. Echo

Commit:
Level 2. Add, List

Commit:
Level 3. Mark as Done

Commit:
Adding Task class and started on task 4

Commit:
Task 3. Mark as Done With Task

Commit:
Level 4. ToDos, Events, Deadlines

Commit:
Level 4. ToDos, Events, Deadlines Using Inheritance

Commit:
Tried to do A-TextUiTesting, but need help

Commit:
Level 6. Delete

Commit:
Automated Text UI Testing Completed

Commit:
Level 5. Handle Errors

Commit:
A-Enums

Commit:
Add data folder and task text

Commit:
Allow code to read from victor.txt

Commit:
Create new victor.txt file is not present

Commit:
Add save function to hard drive file victor.txt

Commit:
Add minimal date recoginization for deadline

Commit:
Add new packages

Commit:
Move all files into Victor package

Commit:
Place all task files into TaskType

Commit:
Refactor the code for the UI and the Storage

Commit:
Rearrange the code into different java files to make it more OOP

Commit:
Cosmetic changes

Commit:
Rearrange some files

Commit:
Divide the classes into packages

Ko-Khan commented on src/main/java/Victor/Parser/Parser.java:
```diff
@@ -0,0 +1,162 @@
+                    ui.displayBarrier();
+                } catch (IndexOutOfBoundsException e) {
+                    ui.displayBarrier();
+                    System.out.println("Oh, you forgot to indicate when is the deadline or maybe you forgot the description.");
```
Maybe consider using line breaks for longer lines?

DominicFuMJ replied:
Fixed. Thanks

Ko-Khan commented on src/main/java/Victor/TaskList/TaskList.java:
```diff
@@ -0,0 +1,49 @@
+
+public class TaskList {
+
+    ArrayList<Task> currentTaskList;
```
I think a plural form to name ArrayLists would be better?

DominicFuMJ replied:
Fixed. Thanks

Ko-Khan commented on src/main/java/Victor/Storage/Storage.java:
```diff
@@ -0,0 +1,76 @@
+        return currentList;
+    }
+
+    public void updateFile(ArrayList<Task> updatedArray) throws IOException{
```
I think you could be more specific here? Like say what is the update.

DominicFuMJ replied:
Added javaDoc.

Ko-Khan commented on src/main/java/Victor/TaskList/TaskList.java:
```diff
@@ -0,0 +1,49 @@
+        this.currentTaskList = new ArrayList<Task>();
+    }
+
+    public void printList() {
```
I like how all method names for this pr are all verbs.

DominicFuMJ replied:
Thanks

Ko-Khan commented on src/main/java/Victor/TaskType/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+public class Deadline extends Task {
+
+    protected String by;
+    protected LocalDate tempBy;
```
It may just be me, but I didn't understand the purpose of this variable from the name.

DominicFuMJ replied:
The tempBy is used as temporary storage of the date, used to hold the date in a LocalDate format, while the by holds the date in a String format. The by is stored in the data file, while the tempBy is normally used to just print out the date in LocalDate format.

Ko-Khan commented on src/main/java/Victor/Storage/Storage.java:
```diff
@@ -0,0 +1,76 @@
+                String[] inputs = nextLine.split("\\|");
+                String fileDataType = inputs[0].trim();
+                switch (fileDataType) {
+                    case "T" -> {
```
Should there be indentation between switch and case?

DominicFuMJ replied:
Fixed.

Ko-Khan reviewed (commented):
I thought the coding quality was great in general

Commit:
Exception Handling. Display error message is data file is missing

Commit:
Merge branch 'add-gradle-support'

Commit:
Add JUnit tests and automate project builds using Gradle

Commit:
Create and moved JAR file

Commit:
Add JavaDocs to majority of the code.

Commit:
Tweak the code to comply with the CS2103 coding standard.

Commit:
Implement a find function to search for words in the description

yuhinarita commented on src/main/java/Victor/Parser/Parser.java:
```diff
@@ -0,0 +1,232 @@
+        String[] inputList = commandLine.split(" ", 2);
+        switch (inputList[0]) {
+        case "list" -> currentTasks.printList();
+        case "mark" -> {
```
Hi! So sorry that my review is this late. I think your coding standards are overall on very high quality, I especially like how descriptive the JavaDocs is. I also like this lambda switch expression, but I think it's only available from Java 12 onwards. So I don't think we can use it in Java 11, as much as I like it ;(

DominicFuMJ replied:
Noted. Thank you

yuhinarita commented on src/main/java/Victor/TaskType/Deadline.java:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    /**
+     * The saveInput is overridden from the Task class. It is used to
```
I think to be consistent with the JavaDocs styles, you might want to start the sentence with a verb like "Returns", "Saves" etc, like you did in other areas of the code.

DominicFuMJ replied:
Noted. Thank you

yuhinarita reviewed (commented):
Overall, there is high quality coding standards. Just keep in mind of the switch-statement with lambda. I don't think it's supported in Java 11, so might want to change to a more conventional style.

Commit:
Upgrade the Event task with LocalDateTime

Commit:
Add checkstyle and fixed any style mistakes

Commit:
Add JavaFX basics

Commit:
Add background color to DialogBox

NeoHW commented on src/main/java/Victor/Storage/Storage.java:
```diff
@@ -0,0 +1,121 @@
+                    currentLists.add(newEvent);
+                }
+                default -> {
+                }
```
perhaps you could consider the option of abstracting this into a method so what it SLAPs!

Commit:
Fix some coding style violations that were listed by the CheckStyle in Gradle.

Fixing these coding style violations should improve readability and ensure that the code is using the same coding style for this project.

Should anyone else make any changes to this project and use the same coding style, it will make it seem that this project has been done by a single person, thus improving the code quality of this project.

Commit:
Add several assertions in the various files.

I made several assertions in order to help check if there are any bugs or errors in the system. I made assertions near the user inputs and areas where the inputs are changed to determine whether the change is correct or if the change is being implemented or not.

With the use of assertions, I hope to be able to better discover errors and bugs within the code and make it easier to rectify it.

Commit:
Add the command to improve the code quality.

The parse method in the Parse file was a very long method, so I decided to finally add the command code to help improve the code quality.

With the command code added, I also managed to implement SLAP as well. This should help if I need to implement the same commands in other parts of the code.

Commit:
Remove dead code from the code files.

There were several pieces of code that were no longer used when I started using the GUI, so I decided to remove it to save space and make it more readable and cleaner.

By removing the dead code, I am helping to improve the code quality of this project. There may still be some dead code that I may have missed, so I will keep an eye out for them

Commit:
Fix some minor bugs

There was a minor bug where the error message would not show if I only have the command and no space. I managed to fix it by sending the entire command line instead of just the rest of the command

Commit:
Add the extension for C-DetectDuplicates for BCD-Extension.

In order to reduce any duplicates within the program, I add the function for the TaskList.java to detect if the new task being added is already inside the list.

I override the equals method in the Deadline, Event and Todo files in order for the ArrayList to properly check if they are the same or not.

For the todo task, I had the arrayList compare only the description.

For the deadline task, I had the arrayList compare the description and the by date.

For the event task, I had the arrayList compare the description, the from date and the to date.

This should prevent any duplicates from being entered into the ArrayList, improving functionality.

DominicFuMJ committed:
Merge pull request #2 from DominicFuMJ/branch-BCD-Extension

Add the extension for C-DetectDuplicates for BCD-Extension.

Commit:
Made some comments in the Command.java file.

For the Command.java file, I made my code based on the code that erv-teo made for his command code. I had to change some of his code to better fit my own code.

Took inspiration about the Command from erv-teo command code.
Made some changes to better fit my code.

Commit:
Merge branch 'branch-BCD-Extension'

Commit:
Merge branch 'master' of https://github.com/DominicFuMJ/ip

Commit:
Adding comment to merge branch.

I originally named the A-Assertions branch differently, so I need to make a new branch and named it properly.

DominicFuMJ committed:
Merge pull request #5 from DominicFuMJ/branch-A-Assertions

Adding comment to merge branch.

Commit:
Create new branch with correct name

Originally named the A-CodeQuality branch differently, so I created a new branch and named it properly

DominicFuMJ committed:
Merge pull request #6 from DominicFuMJ/branch-A-CodeQuality

Create new branch with correct name

Commit:
Create User Guide and did 1 final style check

I created a User Guide that should help guide any users through the functions and commands of Victor. It also helps to explain how to download and run the jar file.

I also did 1 final style check before submission.

Commit:
Add .jar file and link to .jar file.

I created the victor.jar file and added it as a download link to the User Guide. This would allow the users to be able to download the victor.jar file and use victor whenever they want.

DominicFuMJ committed:
Merge pull request #7 from DominicFuMJ/branch-A-UserGuide

Branch a user guide

Commit:
Update .jar link

The download link for the victor.jar didn't work, so I updated the link to be sent to the v0.2 release instead.

Commit:
Change installation instruction.

The installation instruction said that the user must have Java 11, but my code was made to run on Java SDK 17, so I changed it.

Commit:
Undo instruction correction

The instruction correct was incorrect and the previous instruction was correct.

Commit:
Change the switch case for Java 11

Made some changes to the switch case so it works for Java 11.

Commit:
Update the victor.txt file path

I simplify the victor.txt file path so that anyone who tries to use the victor.jar file will be able to create their own victor.jar file on the first try and can access it any time they want.
